moveit_planners/ompl/ompl_interface/src/model_based_planning_context.cpp

```diff
@@ -443,6 +453,64 @@ ompl::base::GoalPtr ompl_interface::ModelBasedPlanningContext::constructGoal()
   return ob::GoalPtr();
 }
 
+ompl::base::PlannerTerminationCondition ompl_interface::ModelBasedPlanningContext::constructPlannerTerminationCondition(
+    double timeout, const ompl::time::point& start)
+{
+  auto it = spec_.config_.find("termination_condition");
+  if (it == spec_.config_.end())
+    return ob::timedPlannerTerminationCondition(timeout - ompl::time::seconds(ompl::time::now() - start));
+  std::string termination_string = it->second;
+  std::vector<std::string> termination_and_params;
+  boost::split(termination_and_params, termination_string, boost::is_any_of("[ ,]"));
+
+  if (termination_and_params.empty())
+    ROS_ERROR_NAMED("model_based_planning_context", "Termination condition not specified");
+  // Terminate if a maximum number of iterations is exceeded or a timeout occurs.
+  // The semantics of "iterations" are planner-specific, but typically it corresponds to the number of times
+  // an attempt was made to grow a roadmap/tree.
+  else if (termination_and_params[0] == "Iteration")
+  {
+    if (termination_and_params.size() > 1)
+      return ob::plannerOrTerminationCondition(
+          ob::timedPlannerTerminationCondition(timeout - ompl::time::seconds(ompl::time::now() - start)),
+          ob::IterationTerminationCondition(std::stoul(termination_and_params[1])));
+    else
+      ROS_ERROR_NAMED("model_based_planning_context", "Missing argument to Iteration termination condition");
+  }
+#if OMPL_VERSION_VALUE >= 1005000
```

Comment:
Add the TODO here about when to remove this (do the cruft doesn't just keep building up over the years)

Reply:
Done.

Reply:
I don't see the second TODO...
When we're about to do a new release, I always grep the codebase for the next and last ros version names and address the TODOs all at once.